language-plutus-core/src/Language/PlutusCore/TypeSynthesis.hs

```diff
@@ -40,25 +37,16 @@ newtype DynamicBuiltinNameTypes = DynamicBuiltinNameTypes
 -- | Configuration of the type checker.
 data TypeConfig = TypeConfig
     { _typeConfigNormalize           :: Bool
-      -- ^ Whether to normalize type annotations
+      -- ^ Whether to normalize type annotations.
     , _typeConfigDynBuiltinNameTypes :: DynamicBuiltinNameTypes
-    }
-
-newtype TypeCheckSt = TypeCheckSt
-    { _gas :: Natural
-    }
-
-data TypeCheckCfg = TypeCheckCfg
-    { _cfgGas        :: Natural     -- ^ Gas to be provided to the typechecker
-    , _cfgTypeConfig :: TypeConfig
+    , _typeConfigGas                 :: Maybe Natural -- ^ The upper limit on the length of type
+                                      -- reductions. If set to 'Nothing', type
+                                      -- reductions will be unbounded.
     }
 
 -- | The type checking monad contains the 'BuiltinTable' and it lets us throw
 -- 'TypeError's.
-type TypeCheckM a = StateT TypeCheckSt (ReaderT TypeConfig (ExceptT (TypeError a) Quote))
-
-gas :: Lens' TypeCheckSt Natural
-gas f s = fmap (\x -> s { _gas = x }) (f (_gas s))
+type TypeCheckM a = ReaderT TypeConfig (ExceptT (TypeError a) Quote)
```

Comment:
We do not charge gas for type checking anymore? Only for type normalization?

Reply:
Yes, that is correct.